scanpy/plotting/_anndata.py

```diff
@@ -1823,6 +1825,7 @@ def _prepare_dataframe(
     if groupby_index is not None:
         # obs_tidy contains adata.obs.index
         # and does not need to be given
+        groupby = groupby[:]  # copy to not modify user passed parameter
```

Comment:
`groupby.copy()` or `list(groupby)` if there's a chance it might not be a list (e.g. a tuple or array)?

Reply:
solved